src/lib/Output/Generator/AbstractFieldTypeHashGenerator.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace Ibexa\Rest\Output\Generator;
+
+use Ibexa\Rest\Output\Generator\Data\DataObjectInterface;
+use Ibexa\Rest\Output\Generator\Json\JsonObject;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Serializer\Exception\ExceptionInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+abstract class AbstractFieldTypeHashGenerator implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private NormalizerInterface $normalizer;
+
+    private bool $strictMode;
+
+    public function __construct(
+        NormalizerInterface $normalizer,
+        ?LoggerInterface $logger = null,
+        bool $strictMode = false
+    ) {
+        $this->normalizer = $normalizer;
+        $this->logger = $logger ?? new NullLogger();
+        $this->strictMode = $strictMode;
+    }
+
+    /**
+     * Generates the field type value $hashValue as a child of the given Object
+     * using $hashElementName as the property name.
+     */
+    public function generateHashValue(
+        DataObjectInterface $parent,
+        string $hashElementName,
+        mixed $hashValue
+    ): void {
+        $parent->$hashElementName = $this->generateValue($parent, $hashValue);
```

Comment:
What's stopping us here from having `\Ibexa\Rest\Output\Generator\Data\DataObjectInterface` a method `setHashElementValue(string $hashElementName, mixed $value)` and using it instead of accessing dynamic property?

Reply:
Not every `DataObjectInterface` has this capacity.

`ArrayObject` can only `set/getName()` and `getParent()` (+ `append()` from actual `ArrayObject` from PHP, and other methods from it).

It's mostly code transfered from JSON generator, that's why it looks funky.